nekton/nii2dcm.py

```diff
@@ -0,0 +1,157 @@
+import os
+from pathlib import Path
+from typing import List
+
+import numpy as np
+import nibabel as nib
+import pydicom
+from pydicom.dataset import FileDataset, Dataset
+import pydicom_seg
+from pydicom_seg.segmentation_dataset import SegmentationDataset
+import SimpleITK as sitk
+
+
+from base import BaseConverter
+from utils.json_helpers import verify_label_dcmqii_json
+
+
+class Nii2DcmSeg(BaseConverter):
+    def __init__(self):
+        super().__init__()
+
+    @staticmethod
+    def _load_segmap(segmentation_map: Path) -> Dataset:
+        """Read the segmentation mapping from the dcmqii standard json
+
+        Args:
+            segmentation_map (Path): Path to the json file
+
+        Returns:
+            Dataset: dataset information extracted from the json
+        """
+        assert os.path.exists(segmentation_map), "Seg mapping `.json` missing"
+
+        assert verify_label_dcmqii_json(
+            segmentation_map
+        ), "Seg mapping `.json` not confirming to DCIM-QII standard, "
+
+        return pydicom_seg.template.from_dcmqi_metainfo(segmentation_map)
+
+    def _check_all_dicoms(self, dcmfiles: List[Path], seg: np.ndarray) -> List[Path]:
+        """Verifies if the number of dicoms and the layers in segmentation match. Also sorts
+            DICOMs based on the instance number.
+
+        Args:
+            dcmfiles (List[Path]): list of path to original dicom files
+            seg (np.ndarray): 3d numpy array of a segmentation
+
+        Returns:
+            List[Path]: sorted list of dicoms based on the order
+        """
+        assert seg.shape[-1] == len(
+            dcmfiles
+        ), f"""Need 1 DICOM per slice of NifTi;
```

Comment:
Is this also the case if there is not segmentation/label for a certain slice? Imagine only one slice has a lesion and it is segmented. Do we need to add layers for all slices even when the rest are empty?

Reply:
ah, ok, this is to match slices in nifti to dcm instances. Note if there is resampling, I do not think we have this case here, but still.

Reply:
Actually, the 1-1 matching was intended for two purposes-

- to prevent resampled images to be matched with original dicoms or vice versa.
- to prevent false matching of slices due to incorrect matching by the user.